src/AwesomizeExpress.re

```diff
@@ -0,0 +1,39 @@
+/* @TODO - Awesomizer needs to support a context and this integration needs
+         to accept a function which can generate that context from the request
+         object.
+   */
+let make = (~schema, ~decoder, ~encoder, ~handler) => {
+  let beltMapToJson = map =>
+    Belt.Map.String.toArray(map) |> Js.Dict.fromArray |> Js.Json.object_;
+  let errorMapToJson = map =>
+    Belt.Map.String.map(map, x =>
+      switch (x) {
+      | None => Js.Json.null
+      | Some(str) => Js.Json.string(str)
+      }
+    )
+    |> beltMapToJson;
+  let awesomize = Awesomize.make(schema);
```

Comment:
This resembles the right march of death that we see in poorly written promise chains.

Reply:
This could probably be made better if the response portion was abstracted out into a function which accepted json and a status code.

Reply:
```reason
let beltMapToJson = map =>
  Belt.Map.String.toArray(map)
  |> Js.Dict.fromArray
  |> Js.Json.object_;

let errorMapToJson= map =>
  Belt.Map.String.map(map, x =>
    switch (x) {
    | None => Js.Json.null
    | Some(str) => Js.Json.string(str)
    }
  )
  |> beltMapToJson;

let sendJson = (res, statusCode, json) =>
  res
  |> Express.Response.status(statusCode)
  |> Express.Response.sendJson(json)
  |> Js.Promise.resolve

let badRequest = Express.Response.StatusCode.BadRequest;
let ok = Express.Response.StatusCode.Ok;

let make = (~schema, ~decoder, ~encoder, ~handler) =>
  (_, req, res) =>
    Express.Request.asJsonObject(req)
    |> Awesomize.make(schema)
    |> Awesomize.fold(
      err => sendJson(res, badRequest, erroMapToJson(err)),
      data =>
        beltMapToJson(data)
        |> decoder
        |> handler(req)
        |> Js.Promise.then_(x => encoder(x) |> Js.Promise.resolve)
        |> Js.Promise.then_(sendJson(res, ok))
    )
    |> Js.Promise.then_(result => result)
```

Reply:
Would like to see the final promise unwrap done away with.

Reply:
I'll update the final value so that it stays promisified, I figured `auth2` was expecting an unwrapped value so I left it as is; I'll update `auth2` accordingly when the time comes.